### PR TITLE
Topic/k1ch/ Introduce API - PUT:/clients/{client_id}

### LINKED ISSUE #13: Implement Client Admin APIs - /clients
# Feature Overview

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

## API Design

Status | URL | Auth | Params | Notes
-- | -- | -- | -- | --
✅   | POST /clients | bearerAdminAuth bearerClientAdminAuth | required Client properties: body | Create a new Client
✅   | GET /clients/:client_id | bearerAdminAuth bearerClientAdminAuth | client_id: path, required | Returns details for the given Client
✅  | DELETE /clients/:client_id | bearerAdminAuth bearerClientAdminAuth | client_id: path, required | Delete the given Client
🔍  | PUT /clients/:client_id | bearerAdminAuth bearerClientAdminAuth | client_id: path, required | Update the given Client
✅  | GET /clients/:client_id/roles | bearerAdminAuth bearerClientAdminAuth | client_id: path, required | Returns a list of roles for the given Client
✅  | POST /clients/:client_id/roles | bearerAdminAuth bearerClientAdminAuth | client_id: path, required Role properties: body | Returns the newly created role for the given Client

### PR DESCRIPTION
**Pull request by k1ch**
# Overview
- [feat: topic/k1ch/ Introduce API PUT:/clients/{client_id}](https://github.com/DMGT-TECH/the-usher-server/commit/ea9b0bebba875a0cd158b4b3e330e5524130b66c)
- [chore: topic/k1ch/api-put-clients/ Add API tests](https://github.com/DMGT-TECH/the-usher-server/pull/99/commits/a5986d5ff16e66fe45ce14795665d48e5dfa6abc)
[chore: topic/k1ch/api-put-clients/ set updated_at value](https://github.com/DMGT-TECH/the-usher-server/pull/99/commits/9f32eddbdf73a8a22aa5ceefb142e4feb1372a3d)

<details open>
  <summary>Added tests: </summary>

image

</details>

<details open>
  <summary>OAS update: </summary>

image

</details> 
## Your PR Checklist 🚨

❤️ Please review the [guidelines for contributing](../docs/CONTRIBUTING.md) to this repository. Our goal is to merge PRs fast 💨 .

- [x] Check your code additions will fail neither code linting checks nor unit tests.
- [x] Additional units tests have been added to prove code updates work and fixes are effective
- [x] Additional documentation has been added (if appropriate)
- [x] Update Assignee field to yourself

## Issue Reference

Closes #13

### CODE REVIEW

File: database/layer/admin-client.js [hunk 1]
```diff
@@ -69,21 +64,35 @@ async function insertClient (tenantName, clientId, name, description, secret) {
...
-    if (results.rowCount === 1) {
-      return 'Update successful'
-    } else {
-      return `Update failed: Client does not exist matching client_id ${clientId}`
-    }
-  } catch (error) {
-    return `Update failed: ${error.message}`
+    const [updatedClient] = await usherDb('clients')
+      .where({ client_id: clientId })
+      .update({
```
Comment (mikelax) [inline database/layer/admin-client.js:83]: @k1ch super trivial, but assuming the `clients` table has an `updated_at` column on it, should we also update that field to `now()` ?

**Review by mikelax — APPROVED** [review]
@k1ch  looks good. I left one trivial question.

File: database/layer/admin-client.js (hunk 1 above)
Comment (k1ch) [reply database/layer/admin-client.js:83]: Good point, yes I will update that.

**Review by mikelax — APPROVED** [review]
:shipit:

**Pull request merged**